clang/docs/ReleaseNotes.rst

```diff
@@ -237,6 +237,8 @@ Bug Fixes in This Version
   when it can affect template argument deduction (#GH122306).
 - Fix crash on code completion of function calls involving partial order of function templates
   (#GH125500).
+- Non-local variable and non-variable declarations in the first clause of a ``for`` loop in C are no longer erroneously
+  considered an error in C23 mode and are allowed as an extension in earlier language modes.
```

Comment:
"_erroneously_ considered an error" kind of sounds tautological, since we're under bug fix section I think "are no longer considered an error" is enough. You may ignore this though, since I'm not an expert in English.

Reply:
Well, I figured that without the ‘erroneously’ it might also be interpreted as the semantics of the language simply having changed rather than communicating that we were just wrong about this, but I don’t think that distinction matters too much, and, yeah, the fact that it’s listed as a bug fix is probably enough to communicate that it’s the latter, so I’ll just remove it.

Reply:
incorrectly